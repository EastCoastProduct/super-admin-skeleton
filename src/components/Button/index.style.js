import { StyleSheet } from 'aphrodite/no-important';
import variables from '../../styles/variables';

const { colors, sizes, transition } = variables;

export default StyleSheet.create({
  button: {
    backgroundColor: colors.input,
    border: `2px solid ${colors.borderAlt}`,
    boxShadow: `1px 1px ${colors.shadow}`,
    borderRadius: sizes.radius,
    color: colors.font,
    fontWeight: 700,
    outline: 'none',
    padding: 10,
    textTransform: 'uppercase',
    transition: transition.base,
    width: '100%',
    ':hover': {
      backgroundColor: colors.background,
    },
    ':active': {
      backgroundColor: colors.header,
      boxShadow: 'none',
    },
    ':disabled': {
      backgroundColor: colors.background,
      border: 0,
      boxShadow: 'none',
      color: colors.shadow,
    },
  },
  empty: {
    background: 0,
    border: 0,
    color: 'inherit',
    padding: 0,
  },
});
