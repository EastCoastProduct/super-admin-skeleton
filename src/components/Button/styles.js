import { StyleSheet } from 'aphrodite/no-important';
import { button } from '../../styles/mixins';

export default StyleSheet.create({
  button: {
    ...button,
  },
  empty: {
    background: 0,
    border: 0,
    color: 'inherit',
    outline: 'none',
    padding: 0,
  },
});
