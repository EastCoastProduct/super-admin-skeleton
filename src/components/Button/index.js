import React, { PropTypes } from 'react';
import { css } from 'aphrodite/no-important';
import styles from './styles';

const Button = ({ className, children, empty, ...rest }) =>
  <button
    className={css(empty ? styles.empty : styles.button,
      className && className)}
    {...rest}
  >{children}</button>;

Button.propTypes = {
  className: PropTypes.object,
  children: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
    PropTypes.element,
  ]).isRequired,
  empty: PropTypes.bool,
};

export default Button;
