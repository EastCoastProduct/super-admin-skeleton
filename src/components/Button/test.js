import React from 'react';
import { shallow } from 'enzyme';
import { StyleSheet } from 'aphrodite/no-important';
import Button from './';

describe('Button component snapshot', () => {
  it('renders basic required data', () => {
    const wrapper = shallow(<Button>Login</Button>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders empty button', () => {
    const wrapper = shallow(<Button empty>Login</Button>);

    expect(wrapper).toMatchSnapshot();
  });

  it('renders with additional styles and attributes', () => {
    const styles = StyleSheet.create({
      additional: {
        margin: 0,
        padding: 0,
      },
    });
    const wrapper = shallow(
      <Button className={styles.additional} disabled type="button">
        Delete
      </Button>
    );

    expect(wrapper).toMatchSnapshot();
  });
});
