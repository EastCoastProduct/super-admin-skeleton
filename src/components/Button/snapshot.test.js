import React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet, StyleSheetTestUtils } from 'aphrodite/no-important';
import Button from './';

describe('Button component snapshot', () => {
  beforeEach(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });
  afterEach(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  it('renders basic required data', () => {
    const tree = renderer.create(<Button>Login</Button>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders empty button', () => {
    const tree = renderer.create(<Button empty>Login</Button>);

    expect(tree.toJSON()).toMatchSnapshot();
  });

  it('renders with additional styles', () => {
    const styles = StyleSheet.create({
      button: {
        margin: 0,
        padding: 0,
      }
    });
    const tree = renderer.create(
      <Button className={styles.button}>Login</Button>
    );

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
