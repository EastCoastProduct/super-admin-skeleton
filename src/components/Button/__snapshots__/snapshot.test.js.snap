exports[`Button component snapshot renders basic required data 1`] = `
<button
  className="button_w1w7v0">
  Login
</button>
`;

exports[`Button component snapshot renders empty button 1`] = `
<button
  className="empty_1b6lvt9">
  Login
</button>
`;

exports[`Button component snapshot renders with additional styles 1`] = `
<button
  className="button_w1w7v0-o_O-button_1433df3">
  Login
</button>
`;
